### PR TITLE
Resolve block not found issue - Sapphire

### PR DESCRIPTION
**Pull request by trizin**
On January 18th, the active flow [failed](https://github.com/oceanprotocol/df-py/actions/runs/7563356604/attempts/2) due to block not found issue originated from Oasis RPC. This PR: 
- Fixes the block not found issue by setting `block_timestamp = 0` thus returning a negative value from `timeSinceTimestamp` so that f(x) != 0
- Verifies that the identified block number is within a 15-minute range of the target timestamp. Throws an error otherwise.

### CODE REVIEW

File: df_py/util/blocktime.py [hunk 1]
```diff
@@ -148,7 +148,29 @@ def timeSinceTimestamp(self, block_i):
...
+            "timestamp_to_block() is returning a block that is too far away"
+        )
+    print(
+        "Returning block number",
+        block_i,
+        "for timestamp",
+        timestamp,
+        "diff",
+        abs(block_timestamp - timestamp),
+    )
```
Comment (trentmc) [inline df_py/util/blocktime.py:173]: Does a unit test capture this corner case?

**Review by trentmc — CHANGES_REQUESTED** [review]

File: df_py/util/blocktime.py [hunk 2]
```diff
@@ -120,7 +120,7 @@ def timeSinceTimestamp(self, block_i):
                 block_timestamp = web3.eth.get_block(int(block_i)).timestamp
             except Exception as e:
                 print(f"An exception occurred while getting block {block_i}, {e}")
-                return 0
+                block_timestamp = 0
             return block_timestamp - self.target_timestamp
```
Comment (trentmc) [inline df_py/util/blocktime.py:124]: I see now that it's ok for this to return `-self.target_timestamp` in the case of an Exception, because it's handled below.

**Review by trentmc — APPROVED** [review]
